packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -278,7 +278,7 @@ class IOSDevice extends Device {
 
       Future<int> launch = runCommandAndStreamOutput(launchCommand, trace: true);
 
-      List<Uri> uris = await launch.then((int result) async {
+      List<Uri> uris = await launch.then<List<Uri>>((int result) async {
```

Comment:
Ah.  Good catch.  I should have flagged this.  Regression.  `flutter analyze` reported:

```
[error] Missing type arguments for generic method 'then<S>'. (packages/flutter_tools/lib/src/commands/daemon.dart, line 467, col 57)
[error] Missing type arguments for generic method 'then<S>'. (packages/flutter_tools/lib/src/ios/devices.dart, line 281, col 37)
```